src/Altinn.App.Api/Controllers/StatelessDataController.cs

```diff
@@ -115,8 +115,12 @@ public class StatelessDataController : ControllerBase
             // runs prefill from repo configuration if config exists
             await _prefillService.PrefillDataModel(owner.PartyId, dataType, appModel);
 
-            Instance virutalInstance = new Instance() { InstanceOwner = owner };
-            await _dataProcessor.ProcessDataRead(virutalInstance, null, appModel);
+            Instance virtualInstance = new Instance() { InstanceOwner = owner };
+            foreach (var dataProcessor in _dataProcessors)
+            {
+                _logger.LogInformation("ProcessDataRead for {modelType} using {dataProcesor}", appModel.GetType().Name, dataProcessor.GetType().Name);
+                await dataProcessor.ProcessDataRead(virtualInstance, null, appModel);
+            }
```

Comment:
Probably not a bad idea. I rejected it because to do it properly I'd need a new service to extract the common parts between stateless and normal.

```c#
internal class DataProcessorWrapper(IEnumerable<IDataProcessor> dataProcessors, ILogger<DataProcessorWrapper> logger)
{
    internal async Task DataProcessRead(...)
    {
        foreach(...)
        {
            // log
            // processRead(...)
        }
    }
    // ProcessDataWrite
}
```

But I did not really like the pattern.